training/training_code/train_ner.py

```diff
@@ -628,6 +628,24 @@ def ftags(name):
     # train confidence branch
     ner_model = train_confidence_branch(ner_model, params, train_data, valid_data, ckpt_dir)
 
+    # generate entity-level metrics for training data
+    print('Computing multi-class metrics for training data')
+    train_scores = generate_mc_metrics(params, ner_model, train_data)
+    print('Training scores:')
+    print(train_scores)
+    csv_out = str(Path(RESULT_DIR, 'train_scores.csv'))
+    train_scores.to_csv(csv_out, index=False)
+    print('Wrote scores to: {}'.format(csv_out))
+
+    # generate entity-level metrics for validation data
+    print('Computing multi-class metrics for validation data')
+    valid_scores = generate_mc_metrics(params, ner_model, valid_data)
+    print('Validation scores:')
+    print(valid_scores)
+    csv_out = str(Path(RESULT_DIR, 'valid_scores.csv'))
+    valid_scores.to_csv(csv_out, index=False)
+    print('Wrote scores to: {}'.format(csv_out))
+
```

Comment:
@xuhdev can we extract this into a for loop? Since we're doing it 3 times over it's repeating quite a bit of code.